### DISCUSSION TITLE
502 Bad Gateway - nginx/1.19.1

### DISCUSSION THREAD

**Discussion by Emroni** [laravel/valet#1132] [Q&A]
Im developing an API in Symfony and needed to upgrade my php version to 7.4. After upgrading I got a 502 whenever I did something wrong (array to string conversion for example), instead of seeing the actual error reporting.

I looked for solutions and tried all the suggestions found both in the issues list in this repo as on StackOverflow; installing, reinstalling, upgrading, brew doctor/cleanup, nginx fastcgi_buffer, all of it. The last on did seem to help on certain errors, but even trying different values did not solve it completely. 

By looking at the log I noticed it was always cut off at 1982 characters, and figured it might have something to do with the logging itself. Finally after setting `fastcgi.logging = 0;` in `php.ini`, the 502 disappeared and I saw normal error reporting. 

Im not sure why this happens when enabling the logging again, but as turning it off isnt really a solution, does anyone have a clue as to why this is happening and what I can do?

Running on:
- valet 2.11.0
- nginx 1.19.1
- php 7.4.8
- symfony 5.1

**Comment by m-ar-c** [comment]
Something similar happened to me (502 errors resolved with `fastcgi.logging = 0`). My guess is that symfony is just sending to much information to nginx. Another way to solve that is to set `error_log = /var/log/php/php_errors.log` in php.ini so that errors are logged to a file instead of being sent through fastcgi (I guess). (If using that, one mustn't forget to create /var/log/php and set ownership and permissions correctly.) Maybe another way would be to still use `fastcgi.logging = 0` but to install a log component in symfony (I didn't test that).
